architecture.tex

```diff
@@ -177,13 +179,13 @@ \subsection{Extensibility}
 SymPy class, generally either \texttt{Basic}, \texttt{Expr}, or
 \texttt{Function}. All SymPy classes used for expression trees~\footnote{Some
   internal classes, such as those used in the polynomial module, do not follow
-  this rule.} should be subclasses of the base class
+  this rule for efficiency reasons.} should be subclasses of the base class
 \texttt{Basic}, which defines some basic methods for symbolic expression
 trees. \texttt{Expr} is the subclass for mathematical expressions that can be
-added and multiplied together. Instances of \texttt{Expr} are typically
+added and multiplied together. Instances of \texttt{Expr} are typically expressions involving
```

Comment:
That's not really what is meant here. What I mean is that the expressions represent some complex number, like `cos(x) + 1` is the complex number cos(x) + 1.